### ISSUE TITLE
macOS build failure in SuperLU on maintenance/1.3.x

### ISSUE THREAD

**Issue by rgommers** [scipy/scipy#10967]
See https://travis-ci.org/scipy/scipy/jobs/601075654:

```
scipy/sparse/linalg/dsolve/SuperLU/SRC/zsp_blas2.c:502:27: note: place parentheses around the '&&' expression to silence this warning

        z_eq(&alpha, &comp_zero) && 

        ~~~~~~~~~~~~~~~~~~~~~~~~~^~

1 warning generated.

clang: scipy/sparse/linalg/dsolve/SuperLU/SRC/zsp_blas3.c

clang: scipy/sparse/linalg/dsolve/SuperLU/SRC/zgssv.c

clang: scipy/sparse/linalg/dsolve/SuperLU/SRC/zutil.c

clang: scipy/sparse/linalg/dsolve/SuperLU/SRC/zgssvx.c

scipy/sparse/linalg/dsolve/SuperLU/SRC/zutil.c:478:1: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]

print_doublecomplex_vec(char *what, int n, doublecomplex *vec)

^

1 warning generated.

scipy/sparse/linalg/dsolve/SuperLU/SRC/zutil.c:478:1: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]

print_doublecomplex_vec(char *what, int n, doublecomplex *vec)

^

1 warning generated.

ar: adding 50 object files to build/temp.macosx-10.6-intel-3.7/libsuperlu_src.a

warning: /Applications/Xcode-9.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: archive library: build/temp.macosx-10.6-intel-3.7/libsuperlu_src.a will be fat and ar(1) will not be able to operate on it

ar: adding 50 object files to build/temp.macosx-10.6-intel-3.7/libsuperlu_src.a

ar: build/temp.macosx-10.6-intel-3.7/libsuperlu_src.a is a fat file (use libtool(1) or lipo(1) and ar(1) on it)

ar: build/temp.macosx-10.6-intel-3.7/libsuperlu_src.a: Inappropriate file type or format

Running from scipy source directory.

/Users/travis/build/scipy/scipy/venv/lib/python3.7/site-packages/numpy/distutils/system_info.py:782: UserWarning: Specified path /usr/local/include/python3.7m is invalid.

  return self.get_paths(self.section, key)

/Users/travis/build/scipy/scipy/venv/lib/python3.7/site-packages/numpy/distutils/system_info.py:782: UserWarning: Specified path /usr/include/python3.7m is invalid.

  return self.get_paths(self.section, key)

/Users/travis/build/scipy/scipy/venv/lib/python3.7/site-packages/numpy/distutils/fcompiler/__init__.py:502: UserWarning: FOPT is used as is, not appended to flags already defined by numpy.distutils! Use NPY_DISTUTILS_APPEND_FLAGS=1 to obtain appending behavior instead (this behavior will become default in a future release).

  flags.extend(getattr(self.flag_vars, tag))

/Users/travis/build/scipy/scipy/venv/lib/python3.7/site-packages/numpy/distutils/fcompiler/__init__.py:517: UserWarning: FFLAGS is used as is, not appended to flags already defined by numpy.distutils! Use NPY_DISTUTILS_APPEND_FLAGS=1 to obtain appending behavior instead (this behavior will become default in a future release).

  fflags = self.flag_vars.flags + dflags + oflags + aflags

/Users/travis/build/scipy/scipy/venv/lib/python3.7/site-packages/numpy/distutils/fcompiler/__init__.py:502: UserWarning: FOPT is used as is, not appended to flags already defined by numpy.distutils! Use NPY_DISTUTILS_APPEND_FLAGS=1 to obtain appending behavior instead (this behavior will become default in a future release).

  flags.extend(getattr(self.flag_vars, tag))

/Users/travis/build/scipy/scipy/venv/lib/python3.7/site-packages/numpy/distutils/fcompiler/__init__.py:517: UserWarning: FFLAGS is used as is, not appended to flags already defined by numpy.distutils! Use NPY_DISTUTILS_APPEND_FLAGS=1 to obtain appending behavior instead (this behavior will become default in a future release).

  fflags = self.flag_vars.flags + dflags + oflags + aflags

/Users/travis/build/scipy/scipy/venv/lib/python3.7/site-packages/numpy/distutils/fcompiler/__init__.py:517: UserWarning: FFLAGS is used as is, not appended to flags already defined by numpy.distutils! Use NPY_DISTUTILS_APPEND_FLAGS=1 to obtain appending behavior instead (this behavior will become default in a future release).

  fflags = self.flag_vars.flags + dflags + oflags + aflags

error: Command "ar rcs build/temp.macosx-10.6-intel-3.7/libsuperlu_src.a build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dldperm.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dmach.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dmemory.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dmyblas2.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dpanel_bmod.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dpanel_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dpivotL.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dpivotgrowth.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dpruneL.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dreadhb.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dreadrb.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dreadtriple.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dsnode_bmod.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dsnode_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dsp_blas2.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dsp_blas3.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dutil.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/dzsum1.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/get_perm_c.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/heap_relax_snode.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/icmax1.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_ccolumn_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_ccopy_to_ucol.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_cdrop_row.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_cpanel_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_cpivotL.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_csnode_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_dcolumn_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_dcopy_to_ucol.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_ddrop_row.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_dpanel_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_dpivotL.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_dsnode_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_heap_relax_snode.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_relax_snode.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_scolumn_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_scopy_to_ucol.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_sdrop_row.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_spanel_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_spivotL.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_ssnode_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_zcolumn_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_zcopy_to_ucol.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_zdrop_row.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_zpanel_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_zpivotL.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/ilu_zsnode_dfs.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/izmax1.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/mark_relax.o build/temp.macosx-10.6-intel-3.7/scipy/sparse/linalg/dsolve/SuperLU/SRC/memory.o" failed with exit status 1

Build failed! (0:02:59.015844 elapsed)
```

**Comment by matthew-brett** [comment]
Sorry if this is obvious, but are you trying to compile for dual architectures (including i386) on a system where the `ar` utility can no longer deal with i386 binaries?   If so, then the fix is to disable dual arch compilation.

**Comment by rgommers** [comment]
That's probably a good guess. I'm not trying anything, it's CI on the 1.3.x branch, which hasn't run in a while.

**Comment by tylerjereddy** [comment]
Duplicate of #10902, recently fixed by Pauli (but I somehow missed that--too many emails!).

Anyway, that's good -- I'll try backporting the fix.

**Comment by tylerjereddy** [comment]
Fixed by backport of #10917 into the maintenance branch.
